README.md

```diff
@@ -28,5 +25,271 @@ We will produce personal data to Confluent Cloud in the following form
+          }
+    }'
+```
+## Registering KEK
+
+We register a KEK so that it can be referenced by encryption rules
+
+```shell
+
+curl --request POST \
+  --url "${SR_URL}/dek-registry/v1/keks" \
+  --header "Authorization: Basic ${SR_BASE64_CRED}" \
+  --header 'Content-Type: application/vnd.schemaregistry.v1+json' \
+  --data "{
+    \"name\": \"${AWS_KEK_NAME}\",
+	  \"kmsType\": \"aws-kms\",
+	  \"kmsKeyId\": \"${AWK_KMS_KEY_ID}\",
+	  \"shared\": true	
+}"
+```
+
+
+## Register Encryption Rules
+
+
+Then, run the following curl to register the rule in Schema Registry. Alternatively, you can do this through the CC UI `Encryption Rules` section.
+
+```shell
+curl --request POST \
+  --url "${SR_URL}/subjects/csfle-kek-shared-demo-value/versions" \
+  --header "Authorization: Basic ${SR_BASE64_CRED}" \
+  --header 'Content-Type: application/vnd.schemaregistry.v1+json' \
+  --data "{
+        \"ruleSet\": {
+        \"domainRules\": [
+      {
+        \"name\": \"encryptPII\",
+        \"kind\": \"TRANSFORM\",
+        \"type\": \"ENCRYPT\",
+        \"mode\": \"WRITEREAD\",
+        \"tags\": [\"PII\"],
+        \"params\": {
+           \"encrypt.kek.name\": \"${AWS_KEK_NAME}\"
+          },
+        \"onFailure\": \"ERROR,NONE\"
+        }
+        ]
+      } 
+    }"
+```
+
+We can check that everything is registered correctly by either executing
+```shell
+curl  --url "${SR_URL}/subjects/csfle-kek-shared-demo-value/versions/latest"   --header "Authorization: Basic ${SR_BASE64_CRED}"  | jq
+```
+
+## Allowing the DEK Registry to access the KMS
+
+Retrieve the permission statements
+
+```shell
+curl  --url "${SR_URL}/dek-registry/v1/policy"  --header "Authorization: Basic ${SR_BASE64_CRED}"  | jq
+{
+  "Sid": "Allow Confluent account(s) (<000000000001>) to use the key",
+  "Effect": "Allow",
+  "Principal": {
+    "AWS": "arn:aws:iam::000000000001:role/confluent-schema-registry-kms-access"
+  },
+  "Action": [
+    "kms:Encrypt",
+    "kms:Decrypt"
+  ],
+  "Resource": "*",
+  "Condition": {
+    "StringEquals": {
+      "aws:PrincipalTag/TenantID": "lsrc-q8okkm"
+    }
+  }
+}
+```
+
+Go back in the KMS key policy configuration on AWS.  Press on "Switch to policy view" button and add permission statements in your AWS KMS policy
+
+![](images/aws_register_confluent_permission_statements.png)
+
+
+## Run the Producer
+
+We continuously produce data with the encryption (the topic `csfle-kek-shared-demo` needs to be created before) by executing
+```
+./gradlew run
+```
+
+We can see in the logs that everything is working fine
+
+```shell
+11:17:33.077 [Thread-0] INFO  KafkaProducer - Kafka Producer started
+11:17:34.495 [kafka-producer-network-thread | producer-1] INFO  KafkaProducer - event produced to csfle-kek-shared-demo
+11:17:36.665 [kafka-producer-network-thread | producer-1] INFO  KafkaProducer - event produced to csfle-kek-shared-demo
+```
+
+or check the encrypted field messages in the CC UI
+
+![](images/cc_ui_consume.png)
```

Comment:
I thought CC UI also has access here (as a client) and would decrypt it automatically